docs/source/notes/extending.rst

```diff
@@ -13,8 +13,11 @@ Extending :mod:`torch.autograd`
 Adding operations to :mod:`~torch.autograd` requires implementing a new
 :class:`Function` subclass for each operation. Recall that :class:`Function` s
 are what :mod:`~torch.autograd` uses to compute the results and gradients, and
-encode the operation history. Every new function requires you to implement 2
-methods:
+encode the operation history. When creating a new :class:`Function`, `ctx` has
+:meth:`~torch.autograd.function._ContextMethodMixin.save_for_backward`,
```

Comment:
These are already mentioned very briefly in the presentation of forward below and link back to the `Function` definition for more details.

I would suggest, instead of this, to add a note just after the definition of forward and backward like:
```
.. note::
  It the user's responsibility to use the special functions in the forward's `ctx` properly to ensure that the new Function works properly with the autograd engine. :meth:`~torch.autograd.function._ContextMethodMixin.save_for_backward` must be used when saving input or ouput of the forward to be used later in the backward. :meth:`~torch.autograd.function._ContextMethodMixin.mark_dirty` must be used to marked any input that is modified inplace by the forward function.
:meth:`~torch.autograd.function._ContextMethodMixin.mark_non_differentiable` must be used to tell the engine if an output is not differentiable.
```
what do you think?

Reply:
I agree :)

Reply:
Will make the changes.